### ISSUE TITLE
workflow_call's github context event_name ambiguity

### ISSUE THREAD

**Issue by Skenvy** [github/docs#16515]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_call

### What part(s) of the article would you like to see updated?

The paragraph for the "workflow_call" event is somewhat ambiguous with the mixing terms of "webhook payload" and "event payload", and says;

> the event payload in the called workflow is the same event payload from the calling workflow

But there is a distinction in naming between the "webhook" payload and the "event" payload across the rest of the docs. When reading this description, it implies that there would be a webhook payload associed with the workflow_call event, perhaps one mentioned on https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads (although there is none) that would simply have the same `github.event` payload as calling workflow, but that the other generic "webhook" payload fields would be equivalent to others.

For example, I assumed upon reading it that the `github.event_name` field, being not in the `github.event` payload, would equal `workflow_call`, and was surprised to discover that the event name was also the same as that of the "webhook payload" that initiated the callee workflow.

This description could be made more clear by saying that it adopts the "webhook" payload of the callee, rather than just the "event" payload.

**Comment by ramyaparimi** [comment]
@Skenvy 
Thanks so much for opening an issue! This needs a technical review, I'll triage this for the team to review :eyes:

**Comment by Altinger-Robert** [comment]
Thank you. 👍

**Comment by Kieudung717** [comment]
Vào 20:50, T.5, 24 Th3, 2022 Nathan Levett ***@***.***> đã
viết:

**Comment by ramyaparimi** [comment]
@Skenvy Thanks so much for your patience. I spoke with the team and we are not going to accept this contribution :yellow_heart:. We really appreciate your time and interest in improving GitHub docs 💖 .

**Comment by Skenvy** [comment]
In other parts of the docs, the `webhook payload`'s fields are synonymous with the `github.*` fields (and the `webhook event payload` is synonymous with the `github.event` field). The only place this distinction appears to be ambiguous is on the description of the workflow_call event. It asserts that the `webhook event payload`, not the entire `webhook payload` is the same as that of the callee (where it appears from dumping the contexts during the called action, that the _entire_ `webhook payload` is the same). Either the description is not wholly unambiguous (not sure why an ambiguity would be ok anyway), or the implementation of the pass-through of the `webhook payload` | `webhook event payload` is wrong (which I assume is not the case). Alternatively the terms are mis-conflated between the `events-that-trigger-workflows` and `webhook-events-and-payloads` pages. Again, not sure why an ambiguity like that is accepted, wilfully so if it's assumed that the names for the payloads across both `events-that-trigger-workflows` and `webhook-events-and-payloads` pages are equivalent and correct.

**Comment by GMNGeoffrey** [comment]
I had exactly the same confusion and this issue appears to be the only documentation on this issue. Please reconsider clarifying what context values are in the case of `workflow_call`

**Comment by RobinHoutevelts** [comment]
I found this issue and a discussion at https://github.com/actions/runner/discussions/1884

I have a workflow `test` that's being dispatched from the ui.
It calls another workflow `create-test-db`.

When `create-test-db` is called from the `test` workflow, `github.event_name` is `workflow_dispatch` instead of `workflow_call`.
I assume this is because `test` was dispatched from the ui, but this is very confusing.

I followed the "solution" in the linked discussion.
My `create-test-db` workflow now has an extra input in the `on.workflow_dispatch.inputs`

```diff
name: create-test-db

on:
  workflow_call:
+    inputs:
+      is_workflow_call:
+        description: 'Ugh'
+        required: true
+        type: string
+        default: 'true'

  workflow_dispatch:

jobs:
```
